src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -188,7 +188,8 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
             if (inputType.ContainsTypeParameter())
             {
                 convertedExpression = expression;
-                if (!hasErrors)
+                // If the expression does not have a constant value, an error will be reported in the caller
```

Comment:
>// If the expression does not have a constant value, an error will be reported in the caller [](start = 16, length = 92)

@agocke It doesn't look like ```private void BuildSwitchLabels(SyntaxList<SwitchLabelSyntax> labelsSyntax, Binder sectionBinder, ArrayBuilder<LabelSymbol> labels, DiagnosticBag tempDiagnosticBag)``` checks for that condition.

Reply:
It feels strange to me that we would be checking for something that is never passed explicitly to ```ClassifyBuiltInConversion``` that was crashing. If type is null, probably we should be checking for that and probably not here.

---
In reply to: [278752470](https://github.com/dotnet/roslyn/pull/35249#discussion_r278752470) [](ancestors = 278752470)

Reply:
BuildSwitchLabels doesn't check directly, but when we build the label we bind the constant pattern, which checks then. This is tested in PatternMatchGenericParameterToMethodGroup